django/db/backends/base/features.py

```diff
@@ -141,6 +141,10 @@ class BaseDatabaseFeatures:
     # Can the backend introspect a TimeField, instead of a DateTimeField?
     can_introspect_time_field = True
 
+    # Some backends may not be able to differentiate BooleanField from other
+    # fields such as IntegerField.
+    introspected_boolean_field_type = 'BooleanField'
```

Comment:
I know there isn't any backwards compat policy on the DB backend interface, but did you check if this will affect any of the major 3rd party ones? :)

Reply:
I'll add a release note as I found usage in some MySQL backends such as https://github.com/nakagami/django-cymysql (copying what Django's MySQL backend does). By the way, this change is reverting back to the approach before c1ae0621bab6b013025ec9024691ce7ad556409e.

Reply:
👌